mass_editing/models/mass_editing_line.py

```diff
@@ -1,51 +1,67 @@
 # Copyright (C) 2019 - Today: GRAP (http://www.grap.coop)
+# Copyright (C) 2020 Iván Todorovich (https://twitter.com/ivantodorovich)
 # @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import api, fields, models
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
 
 
 class MassEditingLine(models.Model):
     _name = "mass.editing.line"
-    _order = "sequence,field_id"
     _description = "Mass Editing Line"
+    _order = "sequence, field_id"
 
     sequence = fields.Integer()
-
-    mass_editing_id = fields.Many2one(comodel_name="mass.editing")
-
+    server_action_id = fields.Many2one(
+        "ir.actions.server",
+        string="Server Action",
+        ondelete="cascade",
+        required=True,
+    )
     model_id = fields.Many2one(
-        comodel_name="ir.model", related="mass_editing_id.model_id"
+        "ir.model",
+        related="server_action_id.model_id",
     )
-
     field_id = fields.Many2one(
-        comodel_name="ir.model.fields",
+        "ir.model.fields",
         string="Field",
-        domain="["
-        "('name', '!=', '__last_update'),"
-        "('ttype', 'not in', ['reference', 'function']),"
-        "('model_id', '=', model_id)"
-        "]",
+        domain="""
+            [
+                ("name", "!=", "__last_update"),
+                ("ttype", "not in", ["reference", "function"]),
+                ("model_id", "=", model_id),
+            ]
+        """,
+        ondelete="cascade",
```

Comment:
This shouldn't be cascade, but `restrict` IMO

Reply:
Wouldn't that prevent module updates that remove fields, for instance?
What's the point on preventing a field removal just for a mass editing rule?

Reply:
The point can be to not allow that unless you remove explicitly the line for being aware, but thinking twice and with the other comment I have put, we can let this way for avoiding harder problems.